go/vt/sqlparser/sql.y

```diff
@@ -3889,6 +3890,26 @@ function_call_keyword:
   {
     $$ = &SubstrExpr{StrVal: NewStrVal($3), From: $5, To: $7}
   }
+| TRIM openb value_expression closeb
+  {
+    $$ = &TrimExpr{Pattern: NewStrVal([]byte(" ")), Str: $3, Dir: NewStrVal([]byte("b"))}
+  }
+| TRIM openb STRING FROM value_expression closeb
```

Comment:
Does MySQL require this to a string literal, or can you use other expressions? Test and seee

Reply:
Yeah, it can take expressions:

```sql
mysql> SELECT TRIM(LEADING concat('x', 'x') FROM 'xxxbarxxx');
+-------------------------------------------------+
| TRIM(LEADING concat('x', 'x') FROM 'xxxbarxxx') |
+-------------------------------------------------+
| xbarxxx                                         |
+-------------------------------------------------+
```